### PR TITLE
[mongoose] Add a missing custom validator and error message param

### PR DESCRIPTION
**Pull request by alex996**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: It's mentioned briefly in the [API docs](https://mongoosejs.com/docs/api.html#schematype_SchemaType-validate), but it's possible to provide an array of `[validator, message]` where `validator` is not only a `() => boolean` but also a `() => Promise<boolean>`. In other words, you can supply an async custom validator into the array as well! Also, `msg` seems to be the older prop, the [currently documented](https://mongoosejs.com/docs/validation.html#custom-validators) one is `message`. Both are [optional](https://github.com/Automattic/mongoose/blob/de3eee5122f462eff41c86fe39dec197d655afed/lib/schematype.js#L681), but `message` should certainly be included in the types.
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

To sum it up, this PR enables two forms of syntax in `Schema.validate`:
1. `[(value: T) =>`**`Promise<boolean>`**`, string]` and
2. `{ validator: () => true,`**`message`**`: 'error' }`

### CODE REVIEW

**Comment by PranavSenthilnathan** [comment]
Congratulations on your first DefinitelyTyped contribution! Thank you for being a part of the community!
